### ISSUE TITLE
Using Multiple swagger files and supporting multiple swagger based micro-services

### ISSUE THREAD

**Issue by grouchal** [APIDevTools/swagger-express-middleware#29]
We have a simple mock server implementation based on swagger-express-middleware:

``` javascript
var myDB = new FileDataStore('.rest-db');
var definitionsPath = 'src/swagger/specifications';
app.use(express.static('www'));
fs.readdir(definitionsPath).forEach(function(filename){
    console.log("Setting up mock services for "+definitionsPath+'/'+filename);
    middleware(definitionsPath+'/'+filename,app, function(err,middleware){
        app.use(
            middleware.files(),
            middleware.metadata(),
            middleware.parseRequest(),
            middleware.validateRequest(),
            middleware.mock(myDB)
        );
    });
});
```

We have found that this implementation does not work as the first swagger file blocks further references to swagger files further down the chain, by returning 404 errors.

**Comment by grouchal** [comment]
A Pull request has been provided with a simple fix in #28

**Comment by margothi** [comment]
This feature would be useful. I'd like to see it added (pull request apparently now #40 )
